daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilParseOutputStreamContentHandler.scala

```diff
@@ -107,6 +183,36 @@ class DaffodilParseOutputStreamContentHandler(out: OutputStream, pretty: Boolean
     outputNewlineStack.push(false)
   }
 
+  private def outputTagName(uri: String, localName: String, qName: String, atts: Maybe[Attributes] = Nope): Unit = {
+    val tagName = {
+      if (qName.nonEmpty) {
+        qName
+      } else {
+        // use attributes
+        lazy val attrs = atts.get
+        lazy val in = attrs.getIndex(uri, localName)
+        if (atts.nonEmpty && attrs.getLength != 0 && in >= 0) {
```

Comment:
Wy are the ``atts.nonEmpty`` && ``attrs.getLength`` checks needed? If ``in >= 0`` that means we found the index and getQName should work, right?